glean-core/src/ping/mod.rs

```diff
@@ -40,7 +41,30 @@ impl PingMaker {
     }
 
     fn get_ping_seq(&self, glean: &Glean, storage_name: &str) -> usize {
-        1
+        // A internal category, not to be touched by anything else
+        let internal_storage = "glean.internal";
+
+        // Sequence numbers are stored as a counter under a name that includes the storage name
+        let seq = CounterMetric::new(CommonMetricData {
+            name: format!("{}#sequence", storage_name),
+            category: "".into(),
+            send_in_pings: vec![internal_storage.into()],
+            lifetime: Lifetime::User,
+            ..Default::default()
+        });
+
+        seq.add(glean, 1);
```

Comment:
What if we move this after the snapshotting below...

Reply:
Yup, consider it done.